src/fosslight_util/download.py

```
@@ -1013,6 +1013,53 @@ def extract_rpm_payload(source_file: str, dest_path: str) -> bool:
     return False
 
 
+def _fix_extracted_permissions(path: str) -> None:
+    try:
+        os.chmod(path, os.stat(path).st_mode | 0o755)
```

Ruff (0.15.15)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)